client/state/jetpack-settings/modules/reducer.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import { merge } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	SERIALIZE,
+	DESERIALIZE,
+	JETPACK_MODULE_ACTIVATE,
+	JETPACK_MODULE_ACTIVATE_FAILURE,
+	JETPACK_MODULE_ACTIVATE_SUCCESS
+} from 'state/action-types';
+
+/**
+ * `Reducer` function which handles request/response actions
+ * concerning Jetpack modules data updates
+ *
+ * @param  {Array}  state  Current state
+ * @param  {Object} action action
+ * @return {Array}         Updated state
+ */
+export const items = ( state = {}, { type, siteId, moduleSlug } ) => {
+	switch ( type ) {
+		case JETPACK_MODULE_ACTIVATE_SUCCESS:
+			return merge( {}, state, {
+				[ siteId ]: {
+					[ moduleSlug ]: {
+						activated: true
+					}
+				}
+			} );
+		// return initial state when serializing/deserializing
+		case SERIALIZE:
+		case DESERIALIZE:
+			return {};
+		default:
+			return state;
+	}
+};
+
+export const initialRequestsState = {
+	activating: {},
+};
+
+/**
+ * `Reducer` function which handles request/response actions
+ * concerning Jetpack modules-related requests
+ *
+ * @param {Object} state - current state
+ * @param {Object} action - action
+ * @return {Object} updated state
+ */
+export const requests = ( state = initialRequestsState, { type, siteId, moduleSlug } ) => {
+	switch ( type ) {
+		case JETPACK_MODULE_ACTIVATE:
+			return merge( {}, state, {
+				activating: {
```

Comment:
I imagined the state tree structure to be indexed by site, then by module, like this:

```
[ siteId ]: {
	[ moduleSlug ]: {
		activating: true
	}
}
```

This way we will not have to duplicate this structure for other module data like `deactivating`, or `enabled` or anything else we might want to store. Because if we keep what you're suggesting here, it would eventually become like:

```
activating: {
	[ siteId ]: {
		[ moduleSlug ]: true
	}
},
deactivating: {
	[ siteId ]: {
		[ moduleSlug ]: false
	}
}
```

while with what I'm suggesting it would avoid the `siteId` and `moduleSlug` duplication:

```
[ siteId ]: {
	[ moduleSlug ]: {
		activating: true,
		deactivating: false
	}
}
```